OpenRA.Mods.Common/Widgets/ButtonWidget.cs

```diff
@@ -245,6 +245,14 @@ public override void Draw()
 
 			var hover = Ui.MouseOverWidget == this || Children.Any(c => c == Ui.MouseOverWidget);
 			DrawBackground(rb, disabled, Depressed, hover, highlighted);
+
+			// Scissor when the text overflows
+			if (textSize.X > Bounds.Width - LeftMargin - RightMargin)
```

Comment:
Wouldn't it be better to introduce a variable here? Because this is used twice.

Reply:
The same pattern is used in `HotkeyEntryWidget` and `TextFieldWidget`. Should I change it there as well?

Reply:
Fixed.